### PR TITLE
Feature/check in

### PR DESCRIPTION
**Pull request by thomasjk0831**
# PR Description
​
Description of what is contained in pull request?  
this PR adds new component for supervisors to make sure guests  are checked-in and on-site during different time intervals

Description of why this change/addition is needed?  
this was needed to allow supervisors to be able to  validate check-in status of the guests at different time intervals during the day as per discussion with the stakeholder. This allows for people on wait list to be given a bed if reserved people don't show up as well having an accurate count of beds taken and record of guests at the shelter that night.
image

Which user story or task is this PR tied to?  
this is related to "show how many beds are left in a given night" - After the supervisor validates that guests are on-site, we can calculate the true number vs the reservation number

How many files and/or lines were changed? 
​4 files changed
​
## Type of Change
​
Types of Change:
- [x] New feature
- [] Refactoring / Bug fix
- [] Documentation update
- [] Executive Change (technical decision change overwrite)
    - Reason for change (required):
​
​
## Testing
​
- [] Yes
- [x] No
- [] Not needed
​
​
## Completeness
Is this feature complete
- [] Feature Complete
- [x] Feature Incomplete
- [] User story/trello card linked here
​
# PR Requirements
- [x] Clean and concise commenting
- [x] Clean and comprehensive documentation included
- [x] Conforms to engineering standards and checks
- [x] There are no merge conflicts
- [x] There are no warnings
- [x] Corresponding documentation elsewhere has been updated to reflect these changes
​
​
## Requester Questions
​
```Add any questions, comments, or concerns you have as the person initiating the pull request for your reviewers```
​
​
## Other
[Rubric](https://www.notion.so/1fc04e4fedeb429ba873b7c68d281707?v=74054da7991341c0bf970f39410c43da)  
[Template Guidelines](https://www.notion.so/Pull-Request-Template-f9264f79e1b649b9845961b5aba3eaff)
​

### CODE REVIEW

**Comment by tajahouse** [comment]
This looks great Thomas! I like that you were able to change the the statuses of the guest check-in and also the hover feature to let the supervisor know what they are checking. That was my only concern until I saw you hover. I just wonder how that would work on a tablet? Hopefully they will just know the order of it.

File: src/components/pages/Guests/Guests.jsx [hunk 1]
```diff
@@ -40,6 +40,7 @@ const Guests = ({}) => {
     axiosWithAuth()
       .get('/members')
       .then(res => {
+        console.log(res.data);
```
Comment (tajahouse) [inline src/components/pages/Guests/Guests.jsx:43]: For the final push, we will need to get rid of the console.logs. I need to do the same so we will do that sometime the end of this week. We can talk about it.

**Review by sadamexx — APPROVED** [review]
Everything looks good. I think what you've built out right now works for now. Hopefully when we have the bed counts and everything else, this will be more helpful to you.

File: src/components/pages/supervisor-pages/SupervisorCheckIn.jsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    let temp = { ...rowData, checked_in: !rowData.checked_in };
+
+    setState({
+      ...state,
+      data: state.data.map(item => {
+        return { ...item, checked_in: !item.checked_in };
+      }),
+    });
+  };
+
```
Comment (tajahouse) [inline src/components/pages/supervisor-pages/SupervisorCheckIn.jsx:73]: This is super clean! Nice!

**Pull request merged**